### PR TITLE
Add support for MySQL 8

### PR DESCRIPTION
**Pull request by lewissteele**
MySQL 8 has switched to caching_sha2_password as its default auth plugin instead of mysql_native_password. `any-db` does not support caching_sha2_password so I've switched out `any-db` for `mysql2` and `pg`. This has fixed the error below that I get when connecting to a MySQL 8 DB

`Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client`

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ if (program.host || configuration?.host) {
     program.host || configuration?.host || "localhost",
     program.user || configuration?.user || "root", // bad practice but unfortunately common, make it easier for the user
     program.password || configuration?.password,
-    program.port || configuration?.port || "3306"
+    program.port || configuration?.port || undefined
```
Comment (joereynolds) [inline src/main.ts]: Why no default fallback?

File: package.json [hunk 2]
```diff
@@ -4,23 +4,20 @@
...
         "@types/node": "^14.0.14",
+        "@types/pg": "^7.14.11",
         "@types/sprintf-js": "^1.1.2",
-        "any-db": "^2.2.1",
-        "any-db-mysql": "^2.1.2",
-        "any-db-postgres": "^2.1.5",
         "commander": "^5.1.0",
         "moo": "^0.5.1",
-        "mysql": "^2.16.0",
+        "mysql2": "^2.2.5",
```
Comment (joereynolds) [inline package.json:13]: Does this support Mysql < 8?

File: docker-compose.yml [hunk 3]
```diff
@@ -10,5 +10,12 @@ services:
       MYSQL_ROOT_PASSWORD: password
     ports:
       - 3306:3306
+  postgres:
```
Comment (joereynolds) [inline docker-compose.yml:13]: <3

File: src/main.ts (hunk 1 above)
Comment (lewissteele) [reply src/main.ts]: `mysql2` and `pg` both fall back to their default ports if it's undefined. The default port for postgres is `5432`, not `3306` 😔

File: package.json (hunk 2 above)
Comment (lewissteele) [reply package.json:13]: I'll check 👍

File: docker-compose.yml (hunk 3 above)
Comment (lewissteele) [reply docker-compose.yml:13]: 😘

File: src/main.ts (hunk 1 above)
Comment (joereynolds) [reply src/main.ts]: Okay so leaving this as undefined is better then I guess? Noice

File: package.json (hunk 2 above)
Comment (joereynolds) [reply package.json:13]: Anything that's >=5.6 is fine

File: src/main.ts (hunk 1 above)
Comment (lewissteele) [reply src/main.ts]: Do you want me to explicitly define the default port for the driver in main.ts instead?

File: package.json (hunk 2 above)
Comment (lewissteele) [reply package.json:13]: Works 👌

File: src/main.ts (hunk 1 above)
Comment (joereynolds) [reply src/main.ts]: I like what you've got leave it as is :+1:

**Pull request merged**